### PR TITLE
v10: Fix build warnings in Web.Common

### PR DESCRIPTION
**Pull request by Zeegaan**
# Notes
- Removed lots of warnings again, i think the interesting stuff is all the redundant "?" checks, cause even though the compiler will give a warning for some of them, its not all that are actually redudant (see the fix test commit, some where failing cause i had removed the ? checks)

- Like in https://github.com/umbraco/Umbraco-CMS/pull/12332 please be aware of the [Obsolete] attrubute, dotnet format adds these when using obsolete methods -.-

# How to test
- Be on v10/dev
- Open your .`globalconfig` file
- On line 44-49 change the severity now to warning instead of suggestion
- Build the project and note the number of build warnings
- Switch branch to this one and build again
- Build warnings should be reduced 🙏

### CODE REVIEW

**Review by nikolajlauridsen — COMMENTED** [review]
Code overall looks good to me 👍, only found a couple of minor things :D Also din't find any new obsolete attributes, so looks like you caught em all :) 

I haven't tested this yet though

**Comment by bjarnef** [comment]
It seems after merging these changing into `v10/contrib` branch I now get an compile error here: https://github.com/umbraco/Umbraco-CMS/blob/v10/contrib/src/Umbraco.Web.Common/Views/UmbracoViewPage.cs#L42

image

In `.globalconfig` it has this on line 43-49:

```
dotnet_analyzer_diagnostic.category-StyleCop.CSharp.DocumentationRules.severity = suggestion
dotnet_analyzer_diagnostic.category-StyleCop.CSharp.ReadabilityRules.severity = suggestion
dotnet_analyzer_diagnostic.category-StyleCop.CSharp.NamingRules.severity = suggestion
dotnet_analyzer_diagnostic.category-StyleCop.CSharp.SpacingRules.severity = suggestion
dotnet_analyzer_diagnostic.category-StyleCop.CSharp.OrderingRules.severity = suggestion
dotnet_analyzer_diagnostic.category-StyleCop.CSharp.MaintainabilityRules.severity = suggestion
dotnet_analyzer_diagnostic.category-StyleCop.CSharp.LayoutRules.severity = suggestion
```

If I change line https://github.com/umbraco/Umbraco-CMS/pull/12349/files#diff-f9c020492385e8ed7a6066d6edafd07b849779a3c533eeb24995235b93f6329bR42 the following instead the code compiles as `ViewData.Model` returns `TModel?`: 

```
TModel? model = ViewData.Model;
```

// cc @bergmania

**Comment by Zeegaan** [comment]
Hey @bjarnef I'm not seeing a compile error on either v10/dev or v10/contrib

image

You've linked to a line, but the link doesn't seem to work for me, just takes me to the changes :(

**Comment by bjarnef** [comment]
@Zeegaan the link should work, but you may need to click "Load diff" at UmbracoViewPage to see the linked line:

image

Strange that it is your screenshot says `TModel` isn't nullable.

**Comment by Zeegaan** [comment]
Really strange indeed, I've just switched to my laptop to test aswell, seems like its also not nullable there 🤔 
When i inspect in Rider i get this: 
image

**Comment by bjarnef** [comment]
Currently I have .NET v6.0.101 installed :)

**Comment by Zeegaan** [comment]
@bjarnef Seems like that is the issue!

I've switched to 6.0.101 and now i get the same 
image

**Comment by bjarnef** [comment]
@Zeegaan strange if it has changed in a patch update of .NET6 🤔

**Comment by bjarnef** [comment]
@Zeegaan okay, maybe it should be mentioned in the contributing guide that it now requires .NET 6.0.2+ 😁 
https://github.com/umbraco/Umbraco-CMS/blob/v10/contrib/.github/CONTRIBUTING.md

// cc @nul800sebastiaan @bergmania

**Comment by nul800sebastiaan** [comment]
Well that is terribly annoying! Fixed: https://github.com/umbraco/Umbraco-CMS/blob/v10/contrib/.github/BUILD.md#debugging-with-vs-code

But also needs to be fixed in the docs for minimum requirements.

File: src/Umbraco.Web.Common/Views/UmbracoViewPage.cs [hunk 1]
```diff
@@ -9,250 +8,254 @@
...
+            }
 
-        private ContentSettings ContentSettings => Context.RequestServices.GetRequiredService<IOptions<ContentSettings>>().Value;
+            _helper = Context.RequestServices.GetRequiredService<UmbracoHelper>();
 
-        private IProfilerHtml ProfilerHtml => Context.RequestServices.GetRequiredService<IProfilerHtml>();
+            if (!(content is null))
+            {
+                _helper.AssignedContentItem = content;
+            }
```
Comment (bjarnef) [inline src/Umbraco.Web.Common/Views/UmbracoViewPage.cs:59]: @Zeegaan couldn't this part be simplified to:

```
TModel model = ViewData.Model;
var content = model as IPublishedContent;
if (content is null && model is IContentModel contentModel)
{
    content = contentModel.Content;
}

_helper = Context.RequestServices.GetRequiredService<UmbracoHelper>();

content ??= UmbracoContext?.PublishedRequest?.PublishedContent;

if (content is not null)
{
    _helper.AssignedContentItem = content;
}
```

Comment (Zeegaan) [reply src/Umbraco.Web.Common/Views/UmbracoViewPage.cs:59]: Hey @bjarnef you are totally right that could be simplified, could i trouble you to make a PR with the changes? I will gladly review it 😄

Comment (bjarnef) [reply src/Umbraco.Web.Common/Views/UmbracoViewPage.cs:59]: @Zeegaan here you go 😁  https://github.com/umbraco/Umbraco-CMS/pull/12528

**Pull request merged**